clients/src/main/java/org/apache/kafka/common/serialization/Deserializer.java

```diff
@@ -32,13 +32,13 @@
     public void configure(Map<String, ?> configs, boolean isKey);
     
     /**
-     *
+     * Deserialize a record value from a bytearray into a value or object.
      * @param topic topic associated with the data
-     * @param data serialized bytes
-     * @return deserialized typed data
+     * @param data serialized bytes; may be null. If null, implementations should return null.
```

Comment:
Generally speaking, it is not guaranteed that the data passed into the serde classes will never be null, and hence when initiating the interface developers should keep in mind to handle that case.

As for Kafka Streams specifically, there are some cases that it should actually guard against null values as they are meaningless, which I think is KAFKA-3836 created for.